gam/gam.py

```diff
@@ -26,10 +36,12 @@ class GAM:
         k (int): number of subpopulations to surface
     """
 
-    def __init__(self, attributions_path="local_attributions.csv", distance="spearman", k=2):
+    def __init__(
+        self, attributions_path="local_attributions.csv", distance="spearman", k=2
+    ):
         self.attributions_path = attributions_path
         self.distance = distance
-        self.k = k
+        self.k = k # Do we want to pass this in here? If so, why include get_optimal as method
```

Comment:
Should we set the default value to `None`, then in cases where a k is not passed in we would use `get_optimal`, otherwise we use the k value passed by the user?

Reply:
Hmm, I like it. I still think we should move `get_optimal` outside of the class (I almost wonder if we should implement it at all), but this seems like a great early way to integrate it into the class.

Reply:
Will do this in my next commit

Reply:
Sounds great - and yeah I think `get_optimal` could easily be a utility outside of the class